pkg/secrets/secrets.go

```diff
@@ -31,132 +30,142 @@ import (
+			if secretResult.Err != nil {
+				log.Fatalln(secretResult.Err)
 			}
 
-			// Iterate until all secrets were returned
-			for range paths {
-				secretResult := parallelReader.ReadSecretResult()
-				err := addSecrets(client, secretResult, secrets)
-				if err != nil {
-					log.Fatalln(err)
-				}
+			err := def.addSecrets(client, secretResult)
+			if err != nil {
+				log.Fatalln(err)
 			}
 		}
-	}
 
-	// Write all secrets to a configured json file
-	if config.SecretPayloadPath != "" {
-		err := writeJSONSecrets(secrets, config.SecretPayloadPath)
-		if err != nil {
-			log.Fatalln(err)
-		}
+		defs = append(defs, def)
 	}
 
-	// Write all secrets to their specified locations
-	if len(locations) != 0 {
-		err := writeSecretsToDestination(secrets, locations)
-		if err != nil {
-			log.Fatalln(err)
+	// output the secret definitions
+	for _, def := range defs {
+		if config.SecretEnv {
+			setEnvSecrets(def.secrets)
 		}
-	}
 
-	// Export secret environment variables if configured, and we are acting as a stub entrypoint for a container
-	if config.SecretEnv {
-		err := setEnvSecrets(secrets)
-		if err != nil {
-			log.Fatalln(err)
+		if def.outputDestination != "" {
+			writeSecretsToDestination(def)
 		}
-	}
-}
-
-// listSecrets returns a list of absolute paths to all secrets located under "secretPath",
-// or an error
-func listSecrets(client *api.Client, secretPath string) ([]string, error) {
-	paths := make([]string, 0, 10)
 
-	list, err := client.Logical().List(secretPath)
-	if err != nil {
-		return nil, fmt.Errorf("there was a problem listing %s: %s", secretPath, err)
-	}
-	if list == nil || len(list.Data) == 0 {
-		return nil, fmt.Errorf("no secrets found under: %s", secretPath)
-	}
-	log.Println("Starting iteration on", secretPath)
-	// list.Data is like: map[string]interface {}{"keys":[]interface {}{"API_KEY", "APPLICATION_KEY", "DB_PASS"}}
-	keys, ok := list.Data["keys"].([]interface{})
-	if !ok {
-		return nil, fmt.Errorf("unexpected list.Data format: %#v", list.Data)
-	}
-	for _, k := range keys {
-		key, ok := k.(string)
-		if !ok {
-			return nil, fmt.Errorf("non-string secret name: %#v", key)
+		if config.SecretPayloadPath != "" {
+			err := writeJSONSecrets(def.secrets, config.SecretPayloadPath)
+			if err != nil {
+				log.Fatalln(err)
+			}
 		}
-		paths = append(paths, path.Join(secretPath, key))
 	}
-
-	return paths, nil
 }
 
-func writeSecretsToDestination(secrets map[string]string, locations map[string]string) error {
-	for secret, secretValue := range secrets {
-		secretDestination, ok := locations[secret]
-		if !ok {
-			continue
-		}
-		err := ioutil.WriteFile(secretDestination, []byte(secretValue), 0600)
+func writeSecretsToDestination(def *SecretDefinition) error {
+	if def.plural {
+		err := writeJSONSecrets(def.secrets, def.outputDestination)
 		if err != nil {
-			return fmt.Errorf("could not write secrets to file '%s': %s", secretDestination, err)
+			return err
```

Comment:
At https://github.com/cruise-automation/daytona/pull/33/files#diff-7b10246886cf200b61f9604cf160d318R149 , we `log.Fatalln(err)`. Why the difference here?

Reply:
`SecretFetcher` doesn't return codes so we fatal there and just return an error here as expected.